src/uk/co/llobile/kotlin/hours/types/FunctionalOption.kt

```diff
@@ -0,0 +1,66 @@
+package uk.co.llobile.kotlin.hours.types
+
+sealed class FunctionalOption<T>: Functor<T>, Applicative<T>, Monad<T> {
+
+    abstract fun isEmpty(): Boolean
+    abstract fun get(): T
+
+    override fun <R> map(fn: (T) -> R): FunctionalOption<R> {
+        return if (isEmpty()) None() else {
```

Comment:
I think _when_ is more elegant that _if/else_

Reply:
Done